test/spec/videoCache_spec.js

```diff
@@ -72,23 +88,54 @@ describe('The video cache', () => {
         <Creatives></Creatives>
       </Wrapper>
     </Ad>
-  </VAST>`,
+  </VAST>`;
+      assertRequestMade({ vastUrl: 'my-mock-url.com' }, expectedValue)
+    });
+
+    it('should make the expected request when store() is called on an ad with a vastPayload', () => {
+      const vastPaload = '<VAST version="3.0"></VAST>';
+      assertRequestMade({ vastPayload: vastPaload }, vastPaload);
+    });
+
+    function assertRequestMade(bid, expectedValue) {
+      store([bid], function() { });
+
+      const request = requests[0];
+      request.method.should.equal('POST');
+      request.url.should.equal('https://prebid.adnxs.com/pbc/v1/cache');
+      request.requestHeaders['Content-Type'].should.equal('text/plain;charset=utf-8');
+
+      JSON.parse(request.requestBody).should.deep.equal({
+        puts: [{
+          type: 'xml',
+          value: expectedValue,
         }],
       });
-    });
+    }
 
-    it('should execute the callback with a successful result when store() is called', () => {
+    function fakeServerCall(bid, responseBody) {
       const callback = sinon.spy();
-      store([ { vastUrl: 'my-mock-url.com' } ], callback);
+      store([ bid ], callback);
       requests[0].respond(
         200,
         {
           'Content-Type': 'application/json',
         },
-        '{"responses":[{"uuid":"c488b101-af3e-4a99-b538-00423e5a3371"}]}');
+        responseBody);
+      return callback;
+    }
+  });
 
-      assertSuccess(callback);
-      callback.firstCall.args[1].should.deep.equal([{ uuid: 'c488b101-af3e-4a99-b538-00423e5a3371' }]);
+  describe('when the cache server breaks its contract', () => {
```

Comment:
aha... thank you. I tested this case in `it('should execute the callback with an error if the cache server responds with malformed JSON')`... forgot to remove it.

Fixed now